### PR TITLE
fix: Add icon to masked textfield component #1536

### LINKED ISSUE #1536: Icon Bug For Textbox Component When Setting Mask Parameter
#### Wave SDK Version, OS

Ecnountered on Wave 0.22.0, Mac OSX 11.3.1

#### Actual behavior

Adding a mask to a textbox component disables the ability to specify an icon.

#### Expected behavior

Be able to add both a mask and icon to a textbox component.

### PR DESCRIPTION
**Pull request by aalencar**
closes #1536

### CODE REVIEW

**Review by mturoci — COMMENTED** [review]
Thanks @aalencar. Seems like there are more missing props though. Let's create a single `Fluent.ITextFieldProps` that will be passed into both `MaskedTextField` and `TextField` to have this resolved once and for all.

File: ui/src/textbox.tsx [hunk 1]
```diff
@@ -70,12 +70,30 @@ export interface Textbox {
...
+      onChange = ({ target }: React.FormEvent<HTMLInputElement | HTMLTextAreaElement>, v?: S) => {
+        v = v || (target as HTMLInputElement).value
 
-      wave.args[m.name] = v ?? (m.value || '')
-      if (m.trigger) wave.push()
-    }
+        wave.args[m.name] = v ?? (m.value || '')
+        if (m.trigger) wave.push()
+      },
+      textFieldProps: Fluent.ITextFieldProps = {
```
Comment (mturoci) [inline ui/src/textbox.tsx:80]: Can we also add `data-test` here?

**Review by mturoci — CHANGES_REQUESTED** [review]
Can we also include `data-test` within `textFieldProps`?

**Comment by aalencar** [comment]
> Can we also include `data-test` within `textFieldProps`?

Let's use `textFieldProps` only for `Fluent.ITextFieldProps` properties.

**Comment by mturoci** [comment]
>Let's use textFieldProps only for Fluent.ITextFieldProps properties.

Why?

**Comment by mturoci** [comment]
That's fine. The goal here is to have a single source of truth for common props.

**Review by mturoci — APPROVED** [review]

**Pull request merged**